core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt

```diff
@@ -0,0 +1,475 @@
+    }
+  }
+
+  /* String content of the note text file given at:
+   *    "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
+   * is displayed in the EditText field (note content area)
+   */
+  private fun displayNote() {
+
+    val noteFile = File("$zimNotesDirectory$articleNoteFileName.txt")
+    if (noteFile.exists()) {
+      writeNote(noteFile)
+    }
+
+    // No action in case the note file for the currently open article doesn't exist
+  }
+
+  private fun writeNote(noteFile: File) {
+    noteFileExists = true
+    val contents = StringBuilder()
+    try {
+      noteFile.bufferedReader().use { bufferReader ->
+        var line: String?
+        while (bufferReader.readLine().also { line = it } != null) {
+          contents.append(line)
+          contents.append(System.getProperty("line.separator"))
+        }
+      }
+    } catch (e: IOException) {
+      e.printStackTrace()
+      Log.d(TAG, "Error reading line with BufferedReader")
+    }
+    add_note_edit_text.setText("$contents") // Display the note content
+    add_note_edit_text.setSelection(add_note_edit_text.text.length - 1)
+    enableShareNoteMenuItem() // As note content exists which can be shared
+    enableDeleteNoteMenuItem()
+  }
+
+  private fun shareNote() {
+
+    /* The note text file corresponding to the currently open article, given at:
+     *    "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
+     * is shared via an app-chooser intent
+     * */
+    if (noteEdited) {
+      // Save edited note before sharing the text file
+      saveNote(add_note_edit_text.text.toString())
+    }
+    val noteFile = File("$zimNotesDirectory$articleNoteFileName.txt")
+    var noteFileUri: Uri? = null
+    if (noteFile.exists()) {
+      noteFileUri = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+        // From Nougat 7 (API 24) access to files is shared temporarily with other apps
+        // Need to use FileProvider for the same
+        FileProvider.getUriForFile(
+          requireContext(), requireContext().packageName + ".fileprovider",
+          noteFile
+        )
+      } else {
+        Uri.fromFile(noteFile)
+      }
+    } else {
+      context.toast(R.string.note_share_error_file_missing, Toast.LENGTH_SHORT)
+    }
+    if (noteFileUri != null) {
+      val noteFileShareIntent = Intent(Intent.ACTION_SEND)
+      noteFileShareIntent.type = "application/octet-stream"
+      noteFileShareIntent.putExtra(Intent.EXTRA_STREAM, noteFileUri)
+      noteFileShareIntent.flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
+      val shareChooser = Intent.createChooser(
+        noteFileShareIntent,
+        getString(R.string.note_share_app_chooser_title)
+      )
+      if (noteFileShareIntent.resolveActivity(requireActivity().packageManager) != null) {
+        startActivity(shareChooser)
+      }
+    }
+  }
+
+  private fun dismissAddNoteDialog() {
+    val dialog = dialog
```

Comment:
there shouldn't be a need to assign dialog to a `val` with the same name

Reply:
done